### ISSUE TITLE
Mistral7b hanging after 2 iterations

### ISSUE THREAD

**Issue by mtairum** [tenstorrent/tt-metal#7540]
On latest commits, Mistral7B is hanging after 2 iterations.

This can be replicated with the mistral7b demo code, on single layer. The hang occurs at `ttnn.to_torch` after model run.

We're currently trying to identify what introduced the regression.

**Comment by mtairum** [comment]
We've tried in two galaxy hosts (single chip) and on a T3000 (single chip) so far.
These are the findings when running the mistral7b demo code (120 iterations):

- T3000 works fine, no hangs on latest main.

- `aus-glx-01` and `aus-glx-08` both hang after 5 iterations even at 1 layer.
    - Tried clean build on localdev, no luck
    - Tried deleting pytest cache and ttnn cache and moving the cache to localdev. No luck
  
One weird issue is that we were seeing in older commits (from ~1 week ago) pytest segfault in the galaxy machines, but no problem on T3000.

Tested several different commits. The process of building was:
- git checkout <commit> 
- git submodule update --recursive
- make nuke
- make build

The default env flags active are:

```
export ARCH_NAME=wormhole_b0
export TT_METAL_HOME=/localdev/${USER}/tt-metal
export PYTHONPATH=/localdev/${USER}/tt-metal
export TT_METAL_ENV=dev
export PYTHONPATH=${PYTHONPATH}:${TT_METAL_HOME}
export CONFIG=release
export TTNN_CONFIG_OVERRIDES='{"enable_fast_runtime_mode": true}'


```

**Comment by mtairum** [comment]
Additionally, we also regenerate the TTNN weight cache files on localdev, but the hang still happens on the aus-glx machines.

**Comment by mtairum** [comment]
On latest main with clean environment the issue seems now concentrated to `aus-glx-01`.

`aus-glx-08` is not showing hangs, with/without program cache for the full model, on latest clean main.

On `aus-glx-01` the hang is on the final dense matmul, which leads to me believe it might be specific to this host.

It is configured like this:

```
self.compute_kernel_config = ttnn.WormholeComputeKernelConfig(
            math_fidelity=ttnn.MathFidelity.HiFi4,
            math_approx_mode=False,
            fp32_dest_acc_en=True,
            packer_l1_acc=True,
        )

...

  self.output_program_config = ttnn.operations.matmul.create_matmul_1d_systolic_array_program_config(
      input_shape_a=ttnn.Shape([1, 1, 32, 4096]),
      input_shape_b=self.output_weight.shape,
      core_grid=args.max_grid_size,
      fp32_dst=args.get_compute_kernel_config().fp32_dest_acc_en,
  )

...
 output = ttnn.linear(
     x,
     self.output_weight,
     program_config=self.output_program_config,
     compute_kernel_config=self.args.get_compute_kernel_config(),
)
```

x.shape = (1,1,32,4096)
weight.shape = (1,1,4096,32000)

Dispatch is using eth cores, so max_core_grid should be (8,8).

I've isolated the above matmul in a unit test, using same program config and dimensions but it passes the test.

**Comment by mtairum** [comment]
Changing from P0 to P1 since we can use other hosts succesfully.

**Comment by uaydonat** [comment]
are we blocked by this? @mtairum

**Comment by mtairum** [comment]
This is a non-blocker. I'll lower to P2.

**Comment by uaydonat** [comment]
@mtairum is this still a problem? Are we seeing a different behaviour on different machines? (it could be di/dt problem, that's why we are interested)

**Comment by mtairum** [comment]
No. On the latest PR for the updated mistral attention (here https://github.com/tenstorrent/tt-metal/pull/8251) we also tested all skipped tests and re-enabled them. All are passing.

As soon as we merge that to main I'll close this issue.

**Comment by mtairum** [comment]
Now merged. https://github.com/tenstorrent/tt-metal/pull/8251
